core/api/src/main/java/com/blazebit/persistence/spi/JpaMetamodelAccessor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2014 - 2018 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.spi;
+
+import javax.persistence.metamodel.Attribute;
+import javax.persistence.metamodel.EntityType;
+import javax.persistence.metamodel.ManagedType;
+import javax.persistence.metamodel.Metamodel;
+
+/**
+ * @author Jan-Willem Gmelig Meyling
+ * @since 1.3.0
+ */
+public interface JpaMetamodelAccessor {
+
+    AttributePath getAttributePath(Metamodel metamodel, ManagedType<?> type, String attributePath);
```

Comment:
You will need java docs for the methods as well

Reply:
Added javadocs here.